lms/services/application_instance.py

```diff
@@ -1,34 +1,41 @@
-from typing import Optional
+from functools import lru_cache
 
 from lms.models import ApplicationInstance
+from lms.services import ConsumerKeyError
 
 
 class ApplicationInstanceService:
-    def __init__(self, db):
+    def __init__(self, db, default_consumer_key):
         self._db = db
+        self._default_consumer_key = default_consumer_key
 
-    def get(self, consumer_key: str) -> Optional[ApplicationInstance]:
-        """Query one ApplicationInstance by consumer_key."""
+    @lru_cache
+    def get(self, consumer_key=None) -> ApplicationInstance:
+        """
+        Return the ApplicationInstance with the given consumer_key.
 
-        return ApplicationInstance.get_by_consumer_key(self._db, consumer_key)
+        If no consumer_key is given return the current request's
+        ApplicationInstance (the ApplicationInstance whose consumer_key matches
+        request.lti_user.oauth_consumer_key).
 
-    @staticmethod
-    def update_settings(ai, canvas_sections_enabled=None, canvas_groups_enabled=None):
```

Comment:
This did some work of hiding the details of how to set particular things. For example you can't just set "canvas.bananas".

On the other hand:

 * Seems annoying to collect here
 * If we want any enforcement the `set()` method looks like the place to me
 * Means this doesn't become a horrible bottle neck through which all settings flow at once